### PR TITLE
Add replace/rename behavior option

### PR DESCRIPTION
**Pull request by rsf123**
Add an option to allow file actions to either replace existing files or use the default behavior of rename.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-    "name": "dcblogdev/laravel-microsoft-graph",
-    "description": "A Laravel Microsoft Graph API (Office365) package",
+    "name": "rsf123/laravel-microsoft-graph",
+    "description": "A Laravel Microsoft Graph API (Office365) package forked from dcblogdev/laravel-microsoft-graph",
```
Comment (dcblogdev) [inline composer.json:3]: no need to change the author details

File: readme.md [hunk 2]
```diff
@@ -21,6 +21,8 @@ API documentation can be found at https://developer.microsoft.com/en-us/graph/do
 
 Please see the [changelog][3] for more information on what has changed recently.
 
+# This fork adds a behavior option to allow replace or rename on creating files and directories.  Usage on createFolder adds an optional behavior after the optional type.  Same on upload.  It will default to "rename"
```
Comment (dcblogdev) [inline readme.md:24]: this should be detailed in a changelog does not need to be in the readme

File: src/MsGraph.php [hunk 3]
```diff
@@ -94,13 +94,20 @@ public function connect($id = null)
                     'token_id' => $result->id,
                     'info' => $me
                 ];
+                
+                //Check if mail is null.  If so, use userPrincialName for email
+                if ($me['mail']===null) {
```
Comment (dcblogdev) [inline src/MsGraph.php:99]: please add formatting:

```
if ($me['mail'] === null) {
    $email = $me['userPrincipalName'];
} else {
    $email = $me['mail'];
}
```

**Comment by dcblogdev** [comment]
once the author changes to composer and readme are removed and formatting applied, I can accept this.

**Review by dcblogdev — CHANGES_REQUESTED** [review]

File: src/MsGraph.php (hunk 3 above)
Comment (rsf123) [reply src/MsGraph.php:99]: Something isn't working properly after I merged in the current version.  I need to do some debugging so need to abandon this request for now

**Pull request closed without merging**